example/index.tsx

```diff
@@ -4,13 +4,33 @@ import * as ReactDOM from 'react-dom';
 import ReactToPrint from '../src/index';
 import ComponentToPrint from './ComponentToPrint';
 
-class Example extends React.Component {
+interface State {
+    text: string;
+}
+
+interface Props {
+
+}
+
+class Example extends React.Component<Props, State> {
     componentRef: ComponentToPrint;
 
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            text: "000000000";
+        };
+    }
+
     handleAfterPrint = () => console.log('after print!');
     handleBeforePrint = () => console.log('before print!');
     renderContent = () => this.componentRef;
     renderTrigger = () => <button type="button">Print this out!</button>;
+    onBeforeGetContent = () => new Promise((resolve, reject) => {
+        this.setState({text: "text changed"}, () => resolve);
+        //if something goes wrong, regect the promise:
```

Comment:
Nit: actually, feel free to remove this comment: I'm going to make a commit after merging this that makes it easier for users to handle the rejection case in both `onBeforePrint` and in `onBeforeGetContent`

Reply:
changed